src/main/java/com/onarandombox/MultiverseCore/destination/LastLocationDestination.java

```diff
@@ -0,0 +1,228 @@
+import com.onarandombox.MultiverseCore.utils.MVPlayerLocation;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.bukkit.util.Vector;
+
+/**
+ * An exact {@link MVDestination}.
+ */
+public class LastLocationDestination implements MVDestination {
+    private String direction = "";
+    private boolean isSpawn;
+    private Location location = null;
+    private Player player = null;
+    private MultiverseWorld world = null;
+    private float yaw = 0;
+
+    //  The internal state of an instance is in a superposition much like
+    //  Schrödinger's cat.  Until observed via getLocation(Entity), the
+    //  instance is both a player's last location in the world and the location
+    //  of the spawn point in the world.  The superposition does not collapse
+    //  into one or the other state until being observed via
+    //  getLocation(Entity).  This is a result of how the MVDestination
+    //  interface is defined.
+    //  (Although, technially it is a superposition of not having the player's
+    //  last location and not having the spawn location until being observed.)
+
+    public static String getID() {
+        return "l";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getIdentifier() {
+        return this.getID();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Vector getVelocity() {
+        return new Vector(0, 0, 0);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isThisType(JavaPlugin plugin, String destination) {
+        if (!(plugin instanceof MultiverseCore)) {
+            return false;
+        }
+
+        // Need one of the following:
+        //   world
+        //   world:direction
+        //   l:world
+        //   l:world:direction
+        String[] items = destination.split(":");
+        if (items.length < 1 || items.length > 3) {
+            return false;
+        }
+        if (items.length == 1) {
+            if (((MultiverseCore) plugin).getMVWorldManager().isMVWorld(items[0])) {
+                // world
+                return true;
+            }
+        } else if (items.length == 2 && ((MultiverseCore) plugin).getMVWorldManager().isMVWorld(items[0])) {
+            // world:direction
+            return true;
+        } else if (items[0].equalsIgnoreCase(this.getIdentifier()) && ((MultiverseCore) plugin).getMVWorldManager().isMVWorld(items[1])) {
+            // l:world  OR  l:world:direction
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Location getLocation(Entity e) {
+        if (!this.isValid() || e == null || !(e instanceof Player)) {
+            return null;
+        }
+
+        player = (Player) e;
+        this.location = MVPlayerLocation.getPlayerLastLocation(player, world);
```

Comment:
Can this location be set in setDestination using a non static method? Such as `plugin.getPlayerLocationManager().getPlayerLastLocation...` or something. Just trying to avoid static stuff like this.